cmd/gardenlet/app/gardenlet.go

```diff
@@ -264,21 +266,22 @@ func NewGardenlet(ctx context.Context, cfg *config.GardenletConfiguration) (*Gar
 
 	var (
 		kubeconfigFromBootstrap []byte
-		csrName                 string
-		seedName                string
+		csrName                 *string
+		seedName                *string
 		err                     error
 	)
 
-	if cfg.GardenClientConnection.KubeconfigSecret != nil {
-		kubeconfigFromBootstrap, csrName, seedName, err = bootstrapKubeconfig(ctx, logger, cfg.GardenClientConnection, cfg.SeedClientConnection.ClientConnectionConfiguration, cfg.SeedConfig)
+	if cfg.GardenClientConnection != nil && cfg.GardenClientConnection.KubeconfigSecret != nil {
```

Comment:
Because it is a pointer and I did not want to just dereference it - I know that it is almost certainly not nil (maybe in the future it is changed) but it does not hurt to include the check?

Reply:
But I do not have a strong opinion. I just also wanted to deal with a value and not a pointer in the other functions

Reply:
I adjusted it so that the pointer is used everywhere